app/lib/package/backend.dart

```diff
@@ -838,6 +840,9 @@ class PackageBackend {
           _incomingBucket.read(tmpObjectName(guid)), filename);
       _logger.info('Examining tarball content ($guid).');
       final sw = Stopwatch()..start();
+      final file = File(filename);
+      final fileBytes = await file.readAsBytes();
```

Comment:
Consider making this streaming.

Reply:
It may work for the hashing, but `fileBytes` may be used later too: when there is an archive file in the canonical bucket, we need to reject the publishing if it differs from the current file. Unless streaming gives us a noticeable benefit, I'd rather keep it as-is.